rsconnect/tests/test_actions.py

```diff
@@ -303,3 +310,89 @@ def test_inspect_environment(self):
         environment = inspect_environment(sys.executable, get_dir("pip1"))
         assert environment is not None
         assert environment.python != ""
+
+
+@pytest.mark.parametrize(
+    ("file_name", "python", "conda_mode", "force_generate", "expected_python", "expected_environment",),
+    [
+        pytest.param(
+            "path/to/file.py",
+            sys.executable,
+            False,
+            False,
+            sys.executable,
+            Environment(
+                conda=None,
+                filename="requirements.txt",
+                locale="en_US.UTF-8",
+                package_manager="pip",
+                source="pip_freeze",
+            ),
+            id="basic",
+        ),
+        pytest.param(
+            "another/file.py",
+            basename(sys.executable),
+            False,
+            False,
+            sys.executable,
+            Environment(
+                conda=None,
+                filename="requirements.txt",
+                locale="en_US.UTF-8",
+                package_manager="pip",
+                source="pip_freeze",
+            ),
+            id="which_python",
+        ),
+        pytest.param(
+            "even/moar/file.py",
+            "whython",
+            True,
+            True,
+            "/very/serious/whython",
+            Environment(
+                conda="/opt/Conda/bin/conda",
+                filename="requirements.txt",
+                locale="en_US.UTF-8",
+                package_manager="pip",
+                source="pip_freeze",
+            ),
+            id="conda_ish",
+        ),
+        pytest.param(
+            "will/the/files/never/stop.py",
+            "argh.py",
+            False,
+            True,
+            "unused",
+            Environment(error="Could not even do things"),
+            id="exploding",
+        ),
+    ],
+)
+def test_get_python_env_info(
+    monkeypatch, file_name, python, conda_mode, force_generate, expected_python, expected_environment,
```

Comment:
nit: prefer `runtime_patch` to `monkeypatch`

Reply:
this is a builtin pytest fixture named `monkeypatch` and I don't see one named `runtime_patch` 😕

Reply:
oh whoops, I thought this was our choice. welp.